### PR TITLE
Refactor the getPodsFor feature

### PR DESCRIPTION
**Pull request by xichengliudui**
Subject
Refactor the getPodsFor feature

Solution
the code is more concise, more efficient

### CODE REVIEW

**Comment by olix0r** [comment]
It appears that there are unit tests failures related to this change: https://github.com/linkerd/linkerd2/pull/6639/checks?check_run_id=3289784935

File: cli/cmd/metrics.go [hunk 1]
```diff
@@ -177,30 +177,16 @@ func getPodsFor(ctx context.Context, clientset kubernetes.Interface, namespace s
...
-			if isOwner(ownerUID, rs.GetOwnerReferences()) {
-				podsRS, err := getPodsFor(ctx, clientset, namespace, fmt.Sprintf("%s/%s", k8s.ReplicaSet, rs.GetName()))
-				if err != nil {
-					return nil, err
-				}
-				pods = append(pods, podsRS...)
-			}
-		}
-		return pods, nil
+		return podList.Items, nil
```
Comment (siggy) [inline cli/cmd/metrics.go:189]: This `isOwner` check is in place to ensure we only select the pods owned by the given Deployment. This change reverts a bug fix made in #5567.

To observe the issue:
```bash
$ curl https://run.linkerd.io/emojivoto.yml | linkerd inject - | kubectl apply -f -
$ curl https://run.linkerd.io/emojivoto-statefulset.yml | linkerd inject - | kubectl apply -f -
$ curl https://run.linkerd.io/emojivoto-daemonset.yml | linkerd inject - | kubectl apply -f -

$ bin/go-run cli diagnostics proxy-metrics -n emojivoto deploy/web | grep "# POD"
# POD web-0 (1 of 3)
# POD web-5f86686c4d-c7td9 (2 of 3)
# POD web-6qsh2 (3 of 3)
```

**Review by siggy — COMMENTED** [review]
we always appreciate more concise and more efficient code! unfortunately this change reverts a subtle bug fix made in #5567. i probably should have left a comment in this part of the code explaining why we're checking for pod ownership in this way.

**Comment by adleong** [comment]
I've created https://github.com/linkerd/linkerd2/pull/6673 to add a test to prevent this kind of regression in the future.  Running this new test on your branch yields:

```console
> go test ./cli
[...]
--- FAIL: TestGetPodsFor (0.01s)
    metrics_test.go:126: ns/pod-1
    metrics_test.go:126: ns/pod-2
    metrics_test.go:128: Expected 1 pod, got 2
FAIL
FAIL    github.com/linkerd/linkerd2/cli/cmd     25.105s
FAIL
```

As in #6623, I think we should close this PR unless you think there are improvements that can be made to this code while maintain the behavior enforced by those new tests.

**Pull request closed without merging**